app/src/main/java/com/kickstarter/viewmodels/ConsentManagementDialogFragmentViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.kickstarter.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.kickstarter.libs.Environment
+import com.kickstarter.ui.SharedPreferenceKey
+import rx.subjects.BehaviorSubject
+
+interface ConsentManagementDialogFragmentViewModel {
+
+    interface Inputs {
+        /** The consent preference of the user represented as a [Boolean] value. True if they allow consent, false if they deny consent. */
+        fun userConsentPreference(consentPreference: Boolean)
+    }
+
+    interface Outputs
+
+    class ConsentManagementDialogFragmentViewModel(environment: Environment) : ViewModel(), Inputs, Outputs {
```

Comment:
` override fun onCleared()` is missing, and the disposable clean up. You can take a look at ProjectRiskViewModel, every time there is a subscription, is added to a composite disposable, and cleared on the `onCleared` method